tests/unit/10_ps-dockerfile.bats

```diff
@@ -116,3 +116,126 @@ teardown() {
+    run bash -c "docker ps -q --no-trunc | grep -q $(< $CID_FILE)"
+    echo "output: "$output
+    echo "status: "$status
+    assert_failure
+  done
+
+  run bash -c "dokku ps:start $TEST_APP"
+  echo "output: "$output
+  echo "status: "$status
+  assert_success
+  for CID_FILE in $DOKKU_ROOT/$TEST_APP/CONTAINER.*; do
+    run bash -c "docker ps -q --no-trunc | grep -q $(< $CID_FILE)"
+    echo "output: "$output
+    echo "status: "$status
+    assert_success
+  done
+
+  run bash -c "dokku ps:restart $TEST_APP"
+  echo "output: "$output
+  echo "status: "$status
+  assert_success
+  for CID_FILE in $DOKKU_ROOT/$TEST_APP/CONTAINER.*; do
+    run bash -c "docker ps -q --no-trunc | grep -q $(< $CID_FILE)"
+    echo "output: "$output
+    echo "status: "$status
+    assert_success
+  done
+
+  run bash -c "dokku ps:rebuild $TEST_APP"
+  echo "output: "$output
+  echo "status: "$status
+  assert_success
+  for CID_FILE in $DOKKU_ROOT/$TEST_APP/CONTAINER.*; do
+    run bash -c "docker ps -q --no-trunc | grep -q $(< $CID_FILE)"
+    echo "output: "$output
+    echo "status: "$status
+    assert_success
+  done
+}
+
+@test "(ps:scale) dockerfile with procfile" {
+  run bash -c "dokku ps:scale $TEST_APP web=2 worker=2"
+  echo "output: "$output
+  echo "status: "$status
+  assert_success
+
+  deploy_app dockerfile-procfile
+  for PROC_TYPE in web worker; do
+    run bash -c "docker ps --format '{{.ID}} {{.Command}}' --no-trunc"
+    echo "output: "$output
+    echo "status: "$status
+    assert_success
+    goodlines=""
+    for CID_FILE in $DOKKU_ROOT/$TEST_APP/CONTAINER.$PROC_TYPE.*; do
+      cid=$(< $CID_FILE)
+      assert_output_contains "$cid"
+      goodlines+=$(echo "$output" | grep "$cid")
+    done
+    output="$goodlines"
+    assert_output_contains "node ${PROC_TYPE}.js" 2
+  done
+
+  run bash -c "dokku ps:scale $TEST_APP web=1 worker=1"
+  echo "output: "$output
+  echo "status: "$status
+  assert_success
+  for PROC_TYPE in web worker; do
+    run bash -c "docker ps --format '{{.ID}} {{.Command}}' --no-trunc"
+    echo "output: "$output
+    echo "status: "$status
+    assert_success
+    goodlines=""
+    for CID_FILE in $DOKKU_ROOT/$TEST_APP/CONTAINER.$PROC_TYPE.*; do
+      cid=$(< $CID_FILE)
+      assert_output_contains "$cid"
+      goodlines+=$(echo "$output" | grep "$cid")
+    done
+    output="$goodlines"
+    assert_output_contains "node ${PROC_TYPE}.js"
+  done
+
+  run bash -c "dokku ps:scale $TEST_APP web=0 worker=0"
+  echo "output: "$output
+  echo "status: "$status
+  assert_success
+  for PROC_TYPE in web worker; do
+    CIDS=""
+    shopt -s nullglob
+    for CID_FILE in $DOKKU_ROOT/$TEST_APP/CONTAINER.$PROC_TYPE.*; do
+      CIDS+=$(< $CID_FILE)
+      CIDS+=" "
+    done
+    shopt -u nullglob
+    run bash -c "[[ -z \"$CIDS\" ]]"
+    echo "output: "$output
+    echo "status: "$status
+    assert_success
+  done
+}
+
+@test "(ps:scale) procfile commands extraction" {
+  deploy_app dockerfile-procfile
```

Comment:
This test seems to be doing double duty given we already test the deployment of this app type in the previous test.

Instead, I recommend running `create_app`, generating a `Procfile` manually at `$DOKKU_ROOT/$TEST_APP/DOKKU_PROCFILE` and then running the `get_cmd_from_procfile()` tests. This way we don't waste the cycles testing something other than the specific thing we want to test. We follow this pattern in the Dockerfile port extraction tests. https://github.com/dokku/dokku/blob/v0.4.13/tests/unit/30_core_ports_2.bats#L100-L124